### PR TITLE
Even more function declaration helpers

### PR DESCRIPTION
**Pull request by FredTheDino**
- Implicit unknown if not specified
- Drive-by: _ctx -> ctx_
- Special case for empty functions
- Fix argument syntax with implicit return
- Solve implicit return if using ->

### CODE REVIEW

**Review by EMattfolk — COMMENTED** [review]
I would like to see some more tests. Maybe something with `fn -> () end` or similar.

**Comment by FredTheDino** [comment]
> I would like to see some more tests. Maybe something with `fn -> () end` or similar.

This test is really messy to add - it's not a parser test, since it's a valid function signature (the `()` is parsed as the return type) - but it won't typecheck since it doesn't return `()` but rather `nil`.

This PR still expects a type after the arrow, followed by statements - so the parsing hasn't changed. For this change to be made we had to do #545 

TLDR: I added the test but it does nothing

**Review by EMattfolk — APPROVED** [review]
